### PR TITLE
DDPB-3915: Apply admin manager rights to app

### PR DESCRIPTION
**Pull request by acsauk**
## Purpose
Now we have an admin manager role we're applying it to different areas of the app to give some more control to certain case managers. The specific permissions for admin managers are:

Admin manager users have permission to:
- Edit deputy email addresses
- Edit their own details
- Edit admin users details (excluding email)
- Discharge deputies
- Create admin users
- Delete admin managers and admins

Admin Manager users don't have permission to:
- Create admin manager or super admins
- Edit details of admin managers or super admins
- View super admin pages

Fixes DDPB-3915

## Approach
I took the opportunity to cover off the areas of the admin app we're updating with elevated admin roles in behat tests. 

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### CODE REVIEW

File: api/config/packages/security.yml [hunk 1]
```diff
@@ -91,7 +91,8 @@ security:
 
     # https://opgtransform.atlassian.net/wiki/spaces/DEPDS/pages/191299647/User+roles
     role_hierarchy:
-        ROLE_SUPER_ADMIN:     [ ROLE_ADMIN, ROLE_ELEVATED_ADMIN ]
+        ROLE_SUPER_ADMIN:     [ ROLE_ADMIN, ROLE_ADMIN_MANAGER ]
```
Comment (jamesrwarren) [inline api/config/packages/security.yml:94]: Question.. as neither role_admin nor role_admin_manager can create admin manager roles, is super admin still getting those permissions from somewhere?

File: api/tests/Behat/behat.yml [hunk 2]
```diff
@@ -242,27 +248,21 @@ v2-tests-goutte:
...
         section-money-out:
             description: Covering the 'Money Out' section of the report
             paths: [ "%paths.base%/features-v2/reporting/sections/money-out" ]
             contexts:
                 - App\Tests\Behat\v2\Reporting\Sections\ReportingSectionsFeatureContext
 
-        section-money-in-high-assets:
-            description: Covering the 'Money in High Assets' section of the report
-            paths: [ "%paths.base%/features-v2/reporting/sections/money-in-high-assets" ]
+        section-visit-care:
```
Comment (jamesrwarren) [inline api/tests/Behat/behat.yml:263]: Ha i've done this too. Alphabetical order is good ;)

File: api/config/packages/security.yml (hunk 1 above)
Comment (jamesrwarren) [reply api/config/packages/security.yml:94]: Cool :)

File: api/tests/Behat/bootstrap/v2/DeputyManagement/DeputyManagementTrait.php [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        }
+
+        $this->interactingWithUserDetails = null;
+        $this->completedFormFields = [];
+    }
+
+    private function emailShouldBeUpdated(string $loggedInUserRole, string $userToBeUpdatedRole): bool
+    {
+        $rolesAdminManagersCanUpdateEmail = [
+            User::ROLE_LAY_DEPUTY,
```
Comment (jamesrwarren) [inline api/tests/Behat/bootstrap/v2/DeputyManagement/DeputyManagementTrait.php:155]: Is this weird?

Comment (jamesrwarren) [reply api/tests/Behat/bootstrap/v2/DeputyManagement/DeputyManagementTrait.php:155]: As in the case of it compared to the others? :)

File: api/tests/Behat/features-v2/admin-management/delete-others.feature [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    When I attempt to delete an existing "admin manager" user
+    Then the user should be deleted
+    When I attempt to delete an existing "admin" user
+    Then the user should be deleted
+
+  @acs
+  Scenario: An admin manager user deletes other admin users
+    Given an admin manager user accesses the admin app
+    And another admin manager user exists
+    When I attempt to delete an existing "admin manager" user
```
Comment (jamesrwarren) [inline api/tests/Behat/features-v2/admin-management/delete-others.feature:18]: i know this is super OCD overload but maybe keep the order of the checks in same order. ie super, manager, admin

File: api/tests/Behat/features-v2/admin-management/edit-others.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
+@v2 @admin-management
```
Comment (jamesrwarren) [inline api/tests/Behat/features-v2/admin-management/edit-others.feature:1]: I feel bad mentioning this as this is already a huuge amount of work and really detailed PR but should we be  validation checks? As in when entering wrong thing in fields, check we get validation messages?

**Review by jamesrwarren — COMMENTED** [review]
Looks amazing. Juts a couple of little bits :)

File: api/tests/Behat/features-v2/admin-management/delete-others.feature (hunk 4 above)
Comment (acsauk) [reply api/tests/Behat/features-v2/admin-management/delete-others.feature:18]: Soooo there was a reason for this and I would normally agree with you wholeheartedly but its to do with setting `$this->interactingWithUser()`. I'm doing this in the `And another admin manager user exists` step so I have the right details when I come to try to delete them. This felt a bit nicer than having  to do some kind of extra step and pass the details between them if that makes sense?

File: api/tests/Behat/features-v2/admin-management/edit-others.feature (hunk 5 above)
Comment (acsauk) [reply api/tests/Behat/features-v2/admin-management/edit-others.feature:1]: We should 😭  I shall add a validation step to the different parts that have it in.

File: api/tests/Behat/features-v2/deputy-management/admin/edit-user.feature [hunk 6]
```diff
@@ -0,0 +1,24 @@
+@v2 @admin-deputy-management
```
[the remark this answers is not in the document]
Comment (acsauk) [reply api/tests/Behat/features-v2/deputy-management/admin/edit-user.feature:1]: Yup yup!

File: api/tests/Behat/bootstrap/v2/DeputyManagement/DeputyManagementTrait.php (hunk 3 above)
Comment (acsauk) [reply api/tests/Behat/bootstrap/v2/DeputyManagement/DeputyManagementTrait.php:155]: Oh wow not sure how I managed that - it's the other roles that are wrong here so I'll update them.

File: api/tests/Behat/features-v2/admin-management/delete-others.feature (hunk 4 above)
Comment (jamesrwarren) [reply api/tests/Behat/features-v2/admin-management/delete-others.feature:18]: Yep totally fine then :)

**Review by jamesrwarren — APPROVED** [review]

**Review by jamesrwarren — APPROVED** [review]

**Pull request merged**